crates/cli/src/codegen/plugin.rs

```diff
@@ -1,147 +1,81 @@
-                    .data(&Default::default()); // Argument is required but not actually used for anything.
-                Ok(str::from_utf8(&import_namespace)?.to_string())
-            }
-        }
-    }
-
-    /// The JS configuration properties supported by this plugin.
-    pub fn config_schema(&self) -> Result<Vec<JsConfigProperty>> {
-        match self {
-            Self::V2 | Self::User { .. } => Ok(vec![]),
-            Self::Default => {
-                let engine = Engine::default();
-                let module = wasmtime::Module::new(&engine, self.as_bytes())?;
-                let mut linker = Linker::new(&engine);
-                wasi_common::sync::snapshots::preview_1::add_wasi_snapshot_preview1_to_linker(
-                    &mut linker,
-                    |s| s,
-                )?;
-                let stdout = WritePipe::new_in_memory();
-                let wasi = WasiCtxBuilder::new()
-                    .inherit_stderr()
-                    .stdout(Box::new(stdout.clone()))
-                    .build();
-                let mut store = wasmtime::Store::new(&engine, wasi);
-                let instance = linker.instantiate(store.as_context_mut(), &module)?;
-                instance
-                    .get_typed_func::<(), ()>(store.as_context_mut(), "config_schema")?
-                    .call(store.as_context_mut(), ())?;
-                drop(store);
-                let mut config_json = vec![];
-                let mut cursor = stdout.try_into_inner().unwrap();
-                cursor.rewind()?;
-                cursor.read_to_end(&mut config_json)?;
-                let config_schema = serde_json::from_slice::<ConfigSchema>(&config_json)?;
-                let mut configs = Vec::with_capacity(config_schema.supported_properties.len());
-                for config in config_schema.supported_properties {
-                    configs.push(JsConfigProperty {
-                        name: config.name,
-                        doc: config.doc,
-                    });
+    pub(crate) fn import_namespace(&self) -> Result<String> {
+        let module = walrus::Module::from_buffer(self.as_bytes())?;
+        let import_namespace: std::borrow::Cow<'_, [u8]> = module
+            .customs
+            .iter()
+            .find_map(|(_, section)| {
+                if section.name() == "import_namespace" {
+                    Some(section)
+                } else {
+                    None
                 }
-                Ok(configs)
-            }
-        }
+            })
+            .ok_or_else(|| anyhow!("Plugin is missing import_namespace custom section"))?
+            .data(&Default::default()); // Argument is required but not actually used for anything.
+        Ok(str::from_utf8(&import_namespace)?.to_string())
     }
 }
 
 /// A validated but uninitialized plugin.
-pub(super) struct UninitializedPlugin<'a> {
+pub struct UninitializedPlugin<'a> {
```

Comment:
Do you need the plugin initialization logic in a crate? I'd prefer to keep that logic in the CLI, it's not a part of the codegen process, it's just used by `javy init-plugin`.

Reply:
I don't need it in the create, lets keep it in the CLI.